lonestar/CMakeLists.txt

```diff
@@ -23,6 +23,36 @@ function(add_test_scale type app)
   endforeach()
 endfunction(add_test_scale)
 
+function(app_cpu PREFIX name bin_name)
+  string(CONCAT target_name ${PREFIX} ${bin_name})
+  add_executable(${target_name} ${name}.cpp)
+  add_dependencies(apps ${target_name})
+  target_link_libraries(${target_name} PRIVATE Galois::shmem lonestar)
+  install(TARGETS ${target_name} DESTINATION "${CMAKE_INSTALL_BINDIR}" COMPONENT apps EXCLUDE_FROM_ALL)
+  set_target_properties(${target_name} PROPERTIES OUTPUT_NAME ${bin_name})
+endfunction(app_cpu)
+
+function(app_gpu PREFIX name bin_name)
+  set(options NO_GPU)
+  set(one_value_args)
+  set(multi_value_args)
+  cmake_parse_arguments(X "${options}" "${one_value_args}" "${multi_value_args}" ${ARGN})
+  string(CONCAT target_name ${PREFIX} ${bin_name})
+  if(${PREFIX} STREQUAL "ANALYTICS_GPU_")
+    add_executable(${target_name} ${name}.cu support.cu)
+  else()
+    add_executable(${target_name} ${name}.cu)
+  endif()
+  install(TARGETS ${target_name} DESTINATION "${CMAKE_INSTALL_BINDIR}" EXCLUDE_FROM_ALL)
+  if(GALOIS_ENABLE_GPU AND NOT ${X_NO_GPU})
+    target_compile_options(${target_name} PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:-w>)
+    target_link_libraries(${target_name} Galois::gpu)
+    set_property(TARGET ${target_name} PROPERTY CUDA_STANDARD 14)
+    set_property(TARGET ${target_name} PROPERTY CUDA_SEPARABLE_COMPILATION ON)
```

Comment:
quick question: is setting `CUDA_SEPARABLE_COMPILATION=on` actually required?

Reply:
Some of the LonestarGPU applications call kernels within the kernel. Enabling this parameter resolves the compilation issues for such applications.

Reply:
Yes, it's required (leads to errors otherwise) for distributed GPU code as well.